testkit/src/main/scala/ch/epfl/scala/bsp/testkit/mock/HappyMockServer.scala

```diff
@@ -259,6 +281,8 @@ class HappyMockServer(base: File) extends AbstractMockServer {
         new SbtBuildTarget("1.0.0", autoImports, scalaBuildTarget, children)
       val cppBuildTarget =
         new CppBuildTarget("C++11", "gcc", "/usr/bin/gcc", "/usr/bin/g++")
```

Comment:
This object is `ch.epfl.scala.bsp4j.CppBuildTarget`, which (in contrast to `ch.epfl.scala.bsp.CppBuildTarget`) uses only strings. Anyway, these are mock values that appear only in tests.

Reply:
I think every path is a String on the type level, but all paths in protocol are described as Uri (where Uri = String). This is true for sources, classpath etc. https://build-server-protocol.github.io/docs/extensions/java.html The bsp client expects uris for fields that contain paths. It might be good way to have os independent paths. For consistency we might want to use Uri as well.

Reply:
I wonder why we need two different classes in the first place. Anyway, this is definitely out of scope for this PR.

Reply:
Changed String to Uri in the `ch.epfl.scala.bsp` package